core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -204,13 +204,20 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
           mod1 >= mod2
       }
 
-      logInfos.sliding(20, 20).foreach { batch =>
+      val tasks: Iterator[Future[_]] = logInfos.grouped(20).map { batch =>
```

Comment:
Thanks! I have updated the PR to add exception handling. I have modified the log message because errors while replaying event logs are already caught in the `mergeApplicationListing` method.